website/static/js/formViewModel.js

```diff
@@ -0,0 +1,45 @@
+/*
+ * Maintains the base class for knockoutJS form ViewModels
+ */
+'use strict';
+
+var ko = require('knockout');
+
+var $osf = require('osfHelpers');
+
+
+var ValidationError = function(errorHeader, errorMessage) {
+    this.errorHeader = errorHeader;
+    this.errorMessage = errorMessage;
+    this.errorLevel = 'warning';
+};
+
+var FormViewModel = function() {
+
+    // Abstract method each ViewModel must impliment. Expected behavior is to either return true
+    // or an array of ValidationError objects
+    self.isValid = function() {
+        throw "Not Implemented";
+    };
+
+    self.submit = function() {
+        try {
+            this.isValid();
+            return true; // Allow form to submit normally
+        } catch (ValidationErrors) {
```

Comment:
This won't work; `ValidationErrors` is not defined. Also, the `try..catch` syntax is incorrect. See the docs here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch

Instead of throwing an array of errors, why not throw an error that can hold multiple messages:

```js
throw new ValidationError(['Username is incorrect.', 'Password is incorrect.]);
```